### ISSUE TITLE
autocomplete-paths is lagging Atom

### ISSUE THREAD

**Issue by ryanaltvater** [atom-community/autocomplete-paths#133]
So I'm experiencing major lag since update because the package is "Rebuilding path cache...". I checked out this page and believe it to be an issue with > 2000 files, but it advises increasing the file count. How do I do that? For now, I've disabled the packaged as it's pretty much unusable. Thanks!

https://github.com/atom-community/autocomplete-paths/wiki/Troubleshooting#maximum-file-limit-exceeded

Atom version: 1.19.1
Package version: 2.6.0

**Comment by linjialiang** [comment]
@ryanaltvater This plugin has lost its role on my computer.
How did you do that

**Comment by ryanaltvater** [comment]
What? Sorry I’m not following what you’re saying/asking.

 

Ryan

**Comment by saschagehlich** [comment]
@ryanaltvater Does your project really contain more than 2000 files?

**Comment by ryanaltvater** [comment]
Yes

Ryan

**Comment by saschagehlich** [comment]
Well then maybe autocomplete-paths is not the right plugin for you. It needs to filter all these files on every keystroke.

It should stop indexing when this message appears though. I'll investigate on that. What OS are you on?

**Comment by ryanaltvater** [comment]
It’s been the right plugin for me for a long time. I’ve been using the package for over a year and this is the first time I’ve experienced this issue after updating both to the latest versions.

Ryan

**Comment by ryanaltvater** [comment]
macOS Sierra 10.12.6

Ryan

**Comment by linjialiang** [comment]
Can you stop indexing when the file is greater than 2000?
Because it's hard to work properly

**Comment by ryanaltvater** [comment]
Can’t really do much when it’s lagging Atom as a whole. So I have disabled it. It lags Atom in a way that seems like it’s continuously trying to index the files but slows Atom down tremendously. So I force-close Atom, open it back up, and disable the package.

Ryan

**Comment by linjialiang** [comment]
@ryanaltvater As long as the problem is solved, I think it will work

**Comment by linjialiang** [comment]
@ryanaltvater Plug-ins are an important part of atom's popularity

**Comment by ryanaltvater** [comment]
What? I’m sorry but I’m having a hard time following some of the things you’re saying/referencing. As long as the problem is solved…because I disabled the plugin? That’s not solving the problem. That’s turning off the problem so that Atom is usable. Solving the problem would be getting the package to work like it did before updating. I could install the previous version, but that means I can’t update it because something is wrong with the latest version…potentially.

Ryan

**Comment by tylkomat** [comment]
I also have a project with more than 2000 files. It seems `autocomplete-paths` is not stopping when it reaches the limit.

Additionally it may help to allow custom ignore settings per project. In my case I have map data which is 17k files. I could add the folder maps to `ignored names`, but then I forget about that and will wonder why it is not working in the next project.

**Comment by mastilver** [comment]
To increase files number: `Atom` -> `Preferences` -> `Packages` -> `Autocomplete Paths` -> `Settings` -> `Max File Count`

For now, I'm downgrading it to version 1: `$ apm install autocomplete-paths@1.0.5` as v2 is almost unusable for me atm :/

@saschagehlich Let me know if I can do anything to help :)

Here's my system config:
```
$ apm -v
apm  1.18.1
npm  3.10.10
node 6.9.5 x64
python 2.7.10
git 2.11.0
```

**Comment by saschagehlich** [comment]
v2.7.0 now properly cancels indexing and hides the status bar item

**Comment by mastilver** [comment]
Can confirm it's working pretty fine now!

Thank you! :)

**Comment by ryanaltvater** [comment]
Even after updating the package to 2.7.0 I still get an error regarding the file count. After doing a little more back-tracking and installing older versions, I noticed that you @saschagehlich introduced 2.0.0 only 4 days ago. No wonder everything "was" working just fine before and all of the sudden it's not. I updated from 1.0.5 to 2.6.0. The indexing issue is introduced in 2.0.0. So now I either stay on 1.0.5 and never update the package, or completely uninstall the package (most likely) because of the introduction of "settings" that breaks the package for me now and throws an error. Even when I increase the file count, it lags Atom pretty hard for me, trying to index all of the files. I guess like you said this just isn't the right plugin for me, as of 4 days ago. Thanks for being the plugin for me up until then. :)
